src/components/app-configure/gasless/AppGasTankDeposit.vue

```diff
@@ -0,0 +1,270 @@
+    const appId = route.params.appId
+    const app = appStore.app(appId)
+    const { data: paymaster } = (
+      await getPaymaster(props.depositTankId, app.network)
+    ).data
+    const web3Provider = new ethers.providers.Web3Provider(walletStore.provider)
+    const wallet = await web3Provider.getSigner()
+    connectedWalletAddress.value = await wallet.getAddress()
+    const owner = selectedGasTank.value.owner
+    const paymasterContract = new ethers.Contract(
+      paymaster.address,
+      paymaster.abi,
+      wallet
+    )
+    const balance = await web3Provider.getBalance(owner)
+    const deposit = await paymasterContract.getBalance(owner)
+    walletBalance.value = ethers.utils.formatEther(balance)
+    walletDeposit.value = ethers.utils.formatEther(deposit)
+  } catch (e) {
+    toast.error('Failed to fetch deposit or balance')
+  } finally {
+    hideLoader()
+  }
+}
+
+function onProceed() {
+  if (props.depositType === 'deposit') depositHandler()
+  if (props.depositType === 'withdraw') withdrawHandler()
+}
+
+async function depositHandler() {
+  showLoader('depositing...')
+  try {
+    const appId = route.params.appId
+    const app = appStore.app(appId)
+    const { data: paymaster } = (
+      await getPaymaster(props.depositTankId, app.network)
+    ).data
+    const web3Provider = new ethers.providers.Web3Provider(walletStore.provider)
+    const wallet = await web3Provider.getSigner()
+    const owner = selectedGasTank.value.owner
+    const paymasterContract = new ethers.Contract(
+      paymaster.address,
+      paymaster.abi,
+      wallet
+    )
+    let tx = await paymasterContract.depositFor(owner, {
+      value: ethers.utils.parseEther(depositAmount.value + ''),
+    })
+    await tx.wait()
+    toast.success('Deposit complete')
+  } catch (e) {
+    if (e?.data?.code === -32000) toast.error('Insufficient funds')
+    else toast.error('Deposit failed')
+  } finally {
+    hideLoader()
+    emits('close')
+  }
+}
+
+async function withdrawHandler() {
+  showLoader('withdrawing...')
+  try {
+    const appId = route.params.appId
+    const app = appStore.app(appId)
+    const { data: paymaster } = (
+      await getPaymaster(props.depositTankId, app.network)
+    ).data
+    const web3Provider = new ethers.providers.Web3Provider(walletStore.provider)
+    const wallet = await web3Provider.getSigner()
+    const owner = selectedGasTank.value.owner
+    const paymasterContract = new ethers.Contract(
+      paymaster.address,
+      paymaster.abi,
+      wallet
+    )
+    let tx = await paymasterContract.withdrawTo(
+      owner,
+      ethers.utils.parseEther(depositAmount.value + '')
+    )
+    await tx.wait()
+    toast.success('Withdraw complete')
+  } catch (e) {
+    console.log(e)
+    toast.error('Withdraw failed')
+  } finally {
+    hideLoader()
+    emits('close')
+  }
+}
+
+async function connectToWallet(chainId: number) {
+  console.log({ chainId })
```

Comment:
Remove this log if not required

Reply:
[done](https://github.com/arcana-network/developer-dashboard/pull/295/commits/0d69c93b4cb98728da38188bdb528894405157d3)